code/obj/item/mob_parts/artifact_parts.dm

```diff
@@ -0,0 +1,540 @@
+					src.name = pick("entwined", "jittery", "soft") + " [src.side]" + " tentacles"
+				else
+					src.name = pick("entwined", "jittery", "pulsing") + " [src.side]" + " leg"
+
+				src.item_state = "martian-limb"
+
+				src.cut_messages = list("swiftly cuts through", "swiftly cut through")
+				src.saw_messages = list("easily saws through", "easily saw through")
+				src.limb_material = "tentacles"
+			if ("precursor")
+				if (src.slot == "l_arm" || src.slot == "r_arm")
+					src.name = pick("ancient", "old", "humming") + " [src.side]" + " device"
+				else
+					src.name =  pick("ancient", "old", "clunky") + " [src.side]" + " leg"
+
+				src.item_state = "precursor-limb"
+
+				src.cut_messages = list("roughly cuts through", "roughly cut through")
+				src.saw_messages = list("messily saws through", "messily saw through")
+				src.limb_material = "metal"
+
+		if (ishuman(new_holder))
+			SPAWN(0.1 SECONDS) // required for abilities to be applied
+				src.on_attach()
+
+	delete()
+		src.on_remove()
+		..()
+
+	remove(show_message)
+		src.on_remove()
+		return ..()
+
+	sever(mob/user)
+		src.on_remove()
+		return ..()
+
+	attach(mob/living/carbon/human/attachee, mob/attacher)
+		if (..())
+			src.on_attach()
+
+	surgery(obj/item/tool)
+		if(remove_stage > 0 && (istype(tool, /obj/item/staple_gun) || istype(tool, /obj/item/suture)) )
+			remove_stage = 0
+
+		else if(remove_stage == 0 || remove_stage == 2)
+			if(istool(tool, TOOL_CUTTING))
+				remove_stage++
+			else
+				return FALSE
+
+		else if(remove_stage == 1)
+			if(istool(tool, TOOL_SAWING))
+				remove_stage++
+			else
+				return FALSE
+
+		if(!isdead(holder))
+			if(prob(40))
+				holder.emote("scream")
+		holder.TakeDamage("chest", 20, 0, 0, DAMAGE_STAB)
+		take_bleeding_damage(holder, tool.the_mob, 15, DAMAGE_STAB)
+
+		switch(remove_stage)
+			if(0)
+				tool.the_mob.visible_message("<span class'alert'>[tool.the_mob] attaches [src.name] to [holder.name] with [tool].</span>", "<span class='alert'>You attach [src.name] to [holder.name] with [tool].</span>")
+				logTheThing("diary", tool.the_mob, holder, "attaches [src.name] to [constructTarget(holder,"combat")].", "combat")
+				logTheThing("combat", tool.the_mob, holder, "attaches [src.name] to [constructTarget(holder,"combat")].")
+			if(1)
+				tool.the_mob.visible_message("<span class='alert'>[tool.the_mob] [src.cut_messages[1]] the [src.limb_material] of [holder.name]'s [src.name] with [tool].</span>", "<span class='alert'>You [src.cut_messages[2]] the [src.limb_material] of [holder.name]'s [src.name] with [tool].</span>")
+			if(2)
+				tool.the_mob.visible_message("<span class='alert'>[tool.the_mob] [src.saw_messages[1]] the [src.limb_material] of [holder.name]'s [src.name] with [tool].</span>", "<span class='alert'>You [src.saw_messages[2]] the [src.limb_material] of [holder.name]'s [src.name] with [tool].</span>")
+
+				SPAWN(rand(15, 20) SECONDS)
+					if(remove_stage == 2)
+						src.remove(FALSE)
+			if(3)
```

Comment:
Consider covering stages 0, 1, 2, 3 to defines to improve readability.  This change is not necessary for this PR.

Reply:
Yeah taking a look at all the other cases might be good for another PR